Neos.Fusion/Classes/Core/Runtime.php

```diff
@@ -298,6 +260,55 @@ public function getLastEvaluationStatus()
         return $this->lastEvaluationStatus;
     }
 
+    /**
+     * todo
+     */
+    public function renderResponse(string $fusionPath, array $contextArray): ResponseInterface
+    {
+        /** Unlike pushContextArray, we don't allow to overrule fusion globals {@see self::pushContext} */
+        foreach ($contextArray as $key => $_) {
+            if ($this->fusionGlobals->has($key)) {
+                throw new Exception(sprintf('Overriding Fusion global variable "%s" via @context is not allowed.', $key), 1706452063);
+            }
+        }
+        $this->pushContextArray($contextArray);
+
+        return $this->withSimulatedLegacyControllerContext(function () use ($fusionPath) {
+            try {
+                $output = $this->render($fusionPath);
+            } catch (RuntimeException $exception) {
+                throw $exception->getWrappedException();
+            } finally {
+                $this->popContext();
+            }
+
+            /**
+             * parse potential raw http response possibly rendered via "Neos.Fusion:Http.Message"
+             * {@see \Neos\Fusion\FusionObjects\HttpResponseImplementation}
+             */
+            $outputStringHasHttpPreamble = is_string($output) && str_starts_with($output, 'HTTP/');
+            if ($outputStringHasHttpPreamble) {
+                return Message::parseResponse($output);
+            }
+
+            if (is_string($output) || $output instanceof \Stringable || $output === null) {
+                return new Response(body: $output);
+            }
+
+            if (is_array($output) || $output instanceof \JsonSerializable || $output instanceof \stdClass || is_bool($output)) {
+                try {
+                    $jsonSerialized = json_encode($output, JSON_THROW_ON_ERROR);
+                } catch (\JsonException $e) {
+                    throw new \RuntimeException(sprintf('Cannot render %s into http response body.', get_debug_type($output)), 1708713158, $e);
+                }
+                $jsonResponse = new Response(body: $jsonSerialized);
+                return $jsonResponse->withHeader('Content-Type', 'application/json');
+            }
```

Comment:
todo revert

Reply:
@kitsunet fyi, i discussed this pr with Wilhelm and we concluded that this magic here will probably not serve very well.

The points were
- it is not expected if not looking into the code that special handling like this exists
  - i would not use it myself but rather be explicit as well
  - there is only a minimal gain for people knowing this feature and a http json response is more obvious and correct and does no harm
- currently the action controller will just silently ignore array values returned from the view, but this would change the behaviour and display accidentally things that are not noticed

thus we throw a nice exception:

> Fusion entry path "root" is expected to render a compatible http response body: string|\Stringable|null. Got array instead.

Reply:
Do we want to support streams here too maybe? Not sure if that can be useful in fusion but who knows... But yeah sounds good to me!

Reply:
Streams are kindof impractical to create and against the fusion cache dna ... implicit support exist because streams are `\Stringable` and utils::streamFor will just return a stream if it finds one.

Still fusion would need a complete rewrite to really support streams